include/mbgl/util/string_indexer.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace mbgl {
+
+using StringIdentity = size_t;
+
+class StringIndexer {
```

Comment:
Would you mind adding a few simple tests for this class in `test/util/string_indexer.test.cpp`?

Reply:
sure. good idea.